Added throwing an error on invalid database name

kamilzych opened the pull request:
Right now there is no database name validation which leads to hard to understand exceptions during code generation when developer will use character in database name which not allowed in Java class name. You can check it by changing database name from "App" to "App.db" in sample application. I've added validation of database name and throwing an Error if validation doesn't pass which should be much easier to understand what exactly is wrong .

agrosner commented:
@kamilzych Nice PR! one thing, can you add checks for if the database name ends in `.db` to throw a different error to say for devs to leave that portion out?

so if they define the name as "somedatabase.db" then it tries to generate:
`somedatabase.db$Database.class`

agrosner commented on compiler/src/main/java/com/raizlabs/android/dbflow/processor/utils/WriterUtils.java:
```diff
@@ -71,4 +74,29 @@ public static boolean writeBaseDefinition(BaseDefinition baseDefinition, Process
 
+    /**
+     * Checks if className is valid Java class name. It will check for Java keywords like abstract, assert, boolean etc.
+     * and it will check if className matches regex pattern [A-Za-z_$]+[a-zA-Z0-9_$]*
```
#req I would say give an example of what it takes to pass the regex, and maybe some failure examples.

agrosner commented:
@kamilzych actually, why should it matter if the databaseName is a java keyword? We don't generate any code that may produce a java keyword. We simply append $Database to the name by default so if it was `void`, it would be `void$Database` as the generated class name.

kamilzych commented:
@agrosner Ok, a made some changes.
- renamed method from 'isValidJavaClassName' to 'isValidDatabaseName' since it no longer checks for java class name validation.
- moved this method from WriterUtils to DatabaseWriter because it's no longer general java class name validation but specific database name validation.
-added javadoc and more clear exception message.
